dotnet/src/Connectors/Connectors.OpenAI/Helpers/OpenAIChatResponseFormatHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text;
+using System.Text.Json;
+using JsonSchemaMapper;
+using OpenAI.Chat;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Helper class to process <see cref="ChatResponseFormat"/> object.
+/// </summary>
+internal static class OpenAIChatResponseFormatHelper
+{
+    /// <summary>
+    /// <see cref="JsonSchemaMapperConfiguration"/> for JSON schema format for structured outputs.
+    /// </summary>
+    private static readonly JsonSchemaMapperConfiguration s_jsonSchemaMapperConfiguration = new()
+    {
+        IncludeSchemaVersion = false,
+        IncludeTypeInEnums = true,
+        TreatNullObliviousAsNonNullable = true,
+        TransformSchemaNode = OpenAIJsonSchemaTransformer.Transform
+    };
+
+    /// <summary>
+    /// Gets instance of <see cref="ChatResponseFormat"/> object for JSON schema format for structured outputs from <see cref="JsonElement"/>.
+    /// </summary>
+    internal static ChatResponseFormat GetJsonSchemaResponseFormat(JsonElement responseFormatElement)
+    {
+        const string DefaultSchemaName = "JsonSchema";
+
+        if (responseFormatElement.TryGetProperty("type", out var typeProperty) &&
+            typeProperty.GetString()?.Equals("json_schema", StringComparison.Ordinal) is true &&
```

Comment:
If type is not json_schema, do we support other types?  Should we throw if we don't support other types and someone specified something else to help them troubleshoot issues?